tools/please_go_test/gotest/find_cover_vars.go

```diff
@@ -69,7 +69,7 @@ func findCoverVars(filepath, importPath string, srcs []string) ([]CoverVar, erro
 		} else if strings.HasSuffix(name, ".go") && !info.IsDir() && !contains(path.Join(dir, name), srcs) {
 			if ok, err := build.Default.MatchFile(dir, name); ok && err == nil {
 				// N.B. The scheme here must match what we do in go_rules.build_defs
-				v := "GoCover_" + strings.Replace(name, ".", "_", -1)
+				v := "GoCover_" + strings.Replace(strings.Replace(name, ".", "_", -1), "-", "_", -1)
```

Comment:
That works, too.  I haven't ventured into this area of the code but was thinking there was a https://golang.org/pkg/go/ast/#File -like struct that could be passed around and used to get the correct `package` name and then there would be a helper function that would filter out invalid characters.  For now, how about:

```go
replacements := []string{
    ".", "_",
    "-", "_",
}
r := strings.NewReplacer(replacements...)
v := "GoCover_" + r.Replace(name)
```

That way future additions are easier to add?

Reply:
I did look at getting the AST earlier on but turns out we don't actually need the package name - they get renamed on import into main. I guess we could use AST parsing to find the names of the cover vars and avoid this - but I'd rather not reparse it really.

Good idea on the Replacer. Not sure I've actually used one of those before...